sysutils/auto-recovery/src/opnsense/scripts/OPNsense/AutoRecovery/auto-recovery.sh

```diff
@@ -0,0 +1,256 @@
+#!/bin/sh
+
+# Copyright (c) 2023 Frank Wall
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+
+# Configuration variables.
+BASE_DIR='/var/etc/auto-recovery'
+ABORT_FILE="${BASE_DIR}/autorecover.abort"
+CONFIG_FILE='/conf/config.xml'
+CONFIG_BACKUP_FILE="${BASE_DIR}/config.xml_recover"
+CONFIG_ORIGINAL_FILE="${BASE_DIR}/config.xml_orig"
```

Comment:
we simply cannot stray from the live config.xml vs. backups procedure adding another layer of misdirection. one thing it will do is confuse the history and I don't see a way to integrate this as a plugin in terms of eventual UX.